rockauth.coffee

```diff
@@ -8,38 +8,47 @@ class @rockauth
 
   @url: (value) ->
     if value
-      @data.set 'rockauth:url', value.replace(/\/$/, '')
-    @data.get 'rockauth:url'
+      @url.value = value.replace /\/$/, ''
+    @url.value
 
   @client_id: (value) ->
     if value
-      @data.set 'rockauth:client_id', value
-    @data.get 'rockauth:client_id'
+      @client_id.value = value
+    @client_id.value
 
   @client_secret: (value) ->
     if value
-      @data.set 'rockauth:client_secret', value
-    @data.get 'rockauth:client_secret'
+      @client_secret.value = value
+    @client_secret.value
 
-  @token: (value) ->
-    if value
-      @data.set 'rockauth:token', value
-    @data.get 'rockauth:token'
+  @authentication: (json) ->
```

Comment:
Let's swap the names of @authentication and @session, so they match the server more. That way, @authentication will be an authentication, and @session will be an authentication + user